### PR TITLE
Feature/annotation changes

### PR DESCRIPTION
**Pull request by PlesnikJakub**
## Summary

Following PR consists of 
* endpoint annotation changes for purpose of simplification and naming endpoint for NSwagClient generator
* class and namespace renaming to follow domain names

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactoring (non-breaking change to code and structure)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Testing

All tests are passing

## Notes

Provided changes will be joined by PR from UI repo with rebuild API Client.

### CODE REVIEW

File: src/WebApi/Application/UseCases/Question/GetInterviewQuestion/GetInterviewQuestionUseCase.cs [hunk 1]
```diff
@@ -9,9 +9,9 @@ public class GetInterviewQuestionUseCase : IGetInterviewQuestionUseCase
 {
     private IOutputPort _outputPort;
 
-    private readonly IInterviewQuestionRepository _interviewQuestionRepository;
+    private readonly IQuestionRepository _interviewQuestionRepository;
```
Comment (TerezaSkutova) [inline src/WebApi/Application/UseCases/Question/GetInterviewQuestion/GetInterviewQuestionUseCase.cs:12]:
```suggestion
    private readonly IQuestionRepository _questionRepository;
```

File: src/WebApi/Application/UseCases/Question/GetInterviewQuestion/GetInterviewQuestionUseCase.cs [hunk 2]
```diff
@@ -9,9 +9,9 @@ public class GetInterviewQuestionUseCase : IGetInterviewQuestionUseCase
...
 
-    public GetInterviewQuestionUseCase(IInterviewQuestionRepository interviewQuestionRepository)
+    public GetInterviewQuestionUseCase(IQuestionRepository interviewQuestionRepository)
```
Comment (TerezaSkutova) [inline src/WebApi/Application/UseCases/Question/GetInterviewQuestion/GetInterviewQuestionUseCase.cs:14]:
```suggestion
    public GetInterviewQuestionUseCase(IQuestionRepository questionRepository)
```

File: src/WebApi/Application/UseCases/Question/UpdateInterviewQuestion/UpdateInterviewQuestionUseCase.cs [hunk 3]
```diff
@@ -8,9 +8,9 @@ public class UpdateInterviewQuestionUseCase : IUpdateInterviewQuestionUseCase
 {
     private IOutputPort _outputPort;
 
-    private readonly IInterviewQuestionRepository _interviewQuestionRepository;
+    private readonly IQuestionRepository _interviewQuestionRepository;
```
Comment (TerezaSkutova) [inline src/WebApi/Application/UseCases/Question/UpdateInterviewQuestion/UpdateInterviewQuestionUseCase.cs:11]:
```suggestion
    private readonly IQuestionRepository _questionRepository;
```

File: src/WebApi/Application/UseCases/Question/GetInterviewQuestion/GetInterviewQuestionUseCase.cs (hunk 1 above)
Comment (TomasNiemczyk98) [reply src/WebApi/Application/UseCases/Question/GetInterviewQuestion/GetInterviewQuestionUseCase.cs]: In doing this change the field name was changed but not it's references, those will also need to be renamed for the code to compile 😄

File: src/WebApi/Infrastructure/Repositories/InterviewQuestionRepository.cs [hunk 4]
```diff
@@ -39,5 +39,20 @@ public async Task<IEnumerable<QuestionModel>> Get(GetInterviewQuestionInput inpu
         var result = await interviewQuestions.ToListAsync();
 
         return _mapper.Map<IEnumerable<QuestionModel>>(result);
+    }
+
+    public Task<IEnumerable<QuestionModel>> GetQuestionsBySetId(int id)
```
Comment (TomasNiemczyk98) [inline src/WebApi/Infrastructure/Repositories/InterviewQuestionRepository.cs:44]: Any reason why this method is not async?

Comment (PlesnikJakub) [reply src/WebApi/Infrastructure/Repositories/InterviewQuestionRepository.cs:44]: there is nothing async in it so it doesn't really make sense.

Comment (PlesnikJakub) [reply src/WebApi/Infrastructure/Repositories/InterviewQuestionRepository.cs:44]: I will change implementation to use FirstOrDefaultAsync instead and make it async 👍

**Pull request merged**